velox/functions/prestosql/geospatial/GeometrySerde.cpp

```diff
@@ -38,4 +38,14 @@ std::string serializeGeometry(const geos::geom::Geometry& geosGeometry) {
   return outputString;
 }
 
+const std::optional<geos::geom::Envelope> getEnvelopeFromGeometry(
+    const StringView& geometry) {
+  std::unique_ptr<geos::geom::Geometry> geosGeometry =
+      geospatial::deserializeGeometry(geometry);
+  if (geosGeometry->isEmpty()) {
+    return std::nullopt;
+  }
+  return *geosGeometry->getEnvelopeInternal();
+}
```

Comment:
Actually, empty geometries will return an empty envelope, which will behave nicely in many cases.  For example, intersection tests will be correct without additional code.

```suggestion
const std::unique_ptr<geos::geom::Envelope> getEnvelopeFromGeometry(
    const StringView& geometry) {
  std::unique_ptr<geos::geom::Geometry> geosGeometry =
      geospatial::deserializeGeometry(geometry);
  return std::unique_ptr<geos::geom::Envelope>(geosGeometry->getEnvelopeInternal());
}
```